### PR TITLE
handle err in metric example test

### PR DESCRIPTION
**Pull request by pkbhowmick**
Related to Issue #4461

### CODE REVIEW

**Review by dmathieu — APPROVED** [review]

File: metric/example_test.go [hunk 1]
```diff
@@ -76,10 +76,18 @@ func ExampleMeter_asynchronous_multiple() {
...
 
 	// This is just a sample of memory stats to record from the Memstats
-	heapAlloc, _ := meter.Int64ObservableUpDownCounter("heapAllocs")
-	gcCount, _ := meter.Int64ObservableCounter("gcCount")
+	heapAlloc, err := meter.Int64ObservableGauge("heapAllocs")
+	if err != nil {
+		fmt.Println("Failed to register gauge for heapAllocs")
+		panic(err)
+	}
+	gcCount, err := meter.Int64ObservableGauge("gcCount")
```
Comment (MrAlias) [inline metric/example_test.go:84]: This changes all of our examples to be for a gauge. I would prefer we change the pretend measured signal to match the instrument so we can showcase all the instrument types.

Comment (utezduyar) [reply metric/example_test.go:84]: I have an idea. An http server that counts the requests and an up/down counter for online people.

Comment (pkbhowmick) [reply metric/example_test.go]: Hey, I updated the example a bit. WDYT now?

Comment (utezduyar) [reply metric/example_test.go]: I think you should put a real web server so people can play around. I also think you should add a real logic to online people. If it turns out too complex, then maybe we should think about another example.

Comment (pkbhowmick) [reply metric/example_test.go]: Hey, I went through the api specifications https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md
It seems the existing example is fine. Just added error handling. consider merging.

**Review by hanyuancheung — APPROVED** [review]

File: metric/example_test.go (hunk 1 above)
Comment (dmathieu) [reply metric/example_test.go]: Runnable examples should be to the point. They should have the fewest boilerplate possible. I think setting up a web server is too much there.

**Review by MrAlias — APPROVED** [review]

**Pull request merged**